### ISSUE TITLE
Make corrections channel selection based on location

### ISSUE THREAD

**Issue by nseidle** [sparkfun/SparkFun_RTK_Everywhere_Firmware#144]
MQTT corrections is currently set by the user (US or EU). The L-Band code has logic to detect device location and auto-selects the correct channel. Use and extend this logic to cover the other IP based territories covered by IP: https://content.u-blox.com/sites/default/files/2024-01/point-perfect-coverage-map.jpg

**Comment by PaulZC** [comment]
This has been given a major facelift. The changes are in release_candidate but have not yet made it into a RC .bin.

The user now selects which geographical region they are in: US, EU, AU, KR or Japan. This is stored in ```settings.geographicRegion```.

The L-Band frequency (for US and EU) is stored in the ```Regional_Information_Table```. We could and probably should use the ```/pp/frequencies/Lb``` MQTT topic to confirm / define these. But we may not have MQTT available out in the field.

The regional correction topics are parsed and saved in settings during ZTP. For IP-only, these are ```/pp/ip/us``` , ```/pp/ip/eu``` , etc.. For L-Band and L-Band+IP, these are ```/pp/Lb/us``` , ```/pp/Lb/eu``` , etc.. The code subscribes to the correct topic based on the geographical region.

The ```Regional_Information_Table``` also includes area definitions as 'squares' based on Lat and Lon. But we're not currently using these. Especially as there is some overlap between KR and Japan. To do the job properly, we'd need polygons. Best and simplest to let the user define which region they are in.

**Comment by nseidle** [comment]
The original functionality should stay in: If the device is L-Band capable, and we are [west of the Atlantic](https://github.com/sparkfun/SparkFun_RTK_Firmware/blob/main/Firmware/RTK_Surveyor/menuPP.ino#L1160), pick US. If we are east, pick EU. New functionality: Don't select L-Band freq until location is known.

New functionality: If the device is using IP corrections, we should attempt to subscribe to the tile in which we are located. If the tile is invalid/unknown, throw a periodic but constant error letting the user know they are outside the service area.

**Comment by PaulZC** [comment]
MQTT_Client has been restructured to make it easier to subscribe to and unsubscribe from topics.

New localized distribution code takes advantage of this. In the following debug messages, the code:

* Subscribes to the ```/pp/Lb/eu``` continental topic - for me in the UK
* Gets a 3D fix
* Subscribes to ```pp/ip/L5N5375W00125/dict```
* Receives the localised tile dict from Thingstream
* Subscribes to ```pp/ip/L5N5375W00125/N5475W00150```
* Unsubscribes from ```/pp/Lb/eu``` and ```pp/ip/L5N5375W00125/dict```

(It's also doing ZTP PROVISIONING at the same time - ignore those debug messages)

(My Blue Peter one-I-prepared-earlier demo failed because I'd got the ZED into a weird state where it was no longer communicating on I2C...!)

```
MQTT_Client subscribing to topic /pp/Lb/eu
Rover Accuracy (m): > 30.000, SIV: 0 GNSS State: No Fix
FreeHeap: 83632 / HeapLowestPoint: 79796 / LargestBlock: 3997684 / Used PSRAM: 141036
STATE_ROVER_NO_FIX --> STATE_ROVER_FIX
MQTT_Client subscribing to topic pp/ip/L5N5375W00125/dict
MQTT_Client subscribing to topic pp/ip/L5N5375W00125/N5475W00150
System time set to: Wednesday, July 03 2024 16:18:11
[ 19794][D][NVM.ino:155] recordSystemSettingsToFileSD(): Removing from SD: /SFE_EVK_Settings_0.txt
[ 19897][D][NVM.ino:174] recordSystemSettingsToFileSD(): Settings recorded to SD: /SFE_EVK_Settings_0.txt
Log file name: /SFE_EVK_240703_161811.ubx
MQTT_Client unsubscribing from topic /pp/Lb/eu
Provisioning: PROVISIONING_WAIT_RTC --> PROVISIONING_NOT_STARTED
MQTT_Client unsubscribing from topic pp/ip/L5N5375W00125/dict
Provisioning: PROVISIONING_NOT_STARTED --> PROVISIONING_CHECK_REMAINING
Log file size: 633 - Generation rate: 0.1kB/s
File loaded from LittleFS: /SFE_EVK_certificate_0.txt
File loaded from LittleFS: /SFE_EVK_privateKey_0.txt
Stored certificates are valid
Days until keys expire: 53
Provisioning: PROVISIONING_CHECK_REMAINING --> PROVISIONING_KEYS_REMAINING
Days until PointPerfect keys expire: 53
PointPerfect keys applied
Provisioning: PROVISIONING_KEYS_REMAINING --> PROVISIONING_WAIT_ATTEMPT
FreeHeap: 83520 / HeapLowestPoint: 79796 / LargestBlock: 3997684 / Used PSRAM: 141284
Rover Accuracy (m): 1.524, SIV: 8 GNSS State: 3D Fix
[ 23973][D][NVM.ino:155] recordSystemSettingsToFileSD(): Removing from SD: /SFE_EVK_Settings_0.txt
[ 24077][D][NVM.ino:174] recordSystemSettingsToFileSD(): Settings recorded to SD: /SFE_EVK_Settings_0.txt
[ 24106][D][NVM.ino:204] recordSystemSettingsToFileLFS(): Removing LittleFS: /SFE_EVK_Settings_0.txt
[ 24340][D][NVM.ino:216] recordSystemSettingsToFileLFS(): Settings recorded to LittleFS: /SFE_EVK_Settings_0.txt
Provisioning: PROVISIONING_WAIT_ATTEMPT --> PROVISIONING_STARTING
Pushing 166 bytes from pp/ip/L5N5375W00125/N5475W00150 topic to GNSS
```

See: b75f937603429fe5f591f102d118e3de1e3aaf9d

TO DO: add localised distribution and the tile level selection to web config

**Comment by PaulZC** [comment]
Resolved in #403
